content/0008-standalone-workflow-execution/rfc.md

```diff
@@ -0,0 +1,294 @@
+have "feature clusters", but they are largely the same size as production and
+are costly, time-wise, to setup, maintain and keep up to date.
+
+Similarly, we have a core repository that contains the components to run
+workflows, but we don't have a way of easily running those workflows in a
+standalone workflow-engine.
+
+## Summary
+
+This RFC defines an extension to the Relay CLI that wraps existing container and
+Kubernetes management technologies to setup a local Kubernetes cluster that runs
+the necessary Relay services for executing workflows in a standalone
+environment. This RFC also defines a secondary mechanism to automatically run
+workflows after bootstrapping has completed. This will allow the tool to be used
+for setting up and managing a full-stack development environment on a local
+machine.
+
+## Motivation
+
+The ability to run workflows in a standalone system allows developers of
+workflows (Relay users) and Puppet developers who work on the Relay Saas to
+reduce churn when testing new code. For the latter, we want to avoid the wait
+time for building and pushing new changes to a remote testing cluster, which
+requires a cycle through Continuous Integration (CI).
+
+## Product-level explanation
+
+This next section will describe the versions of a tool used to create, manage
+and destroy development clusters.
+
+### V1
+
+We add a new set of commands to the [Relay
+CLI](https://github.com/puppetlabs/relay) used to manage a local Kubernetes
+cluster running the core components that execute workflows.
+
+We also generalize workflow translation and validation code currently in
+relay-api into [relay-sdk](https://github.com/puppetlabs/relay-sdk) so it can be
+used by the new cli commands.
+
+#### Relay CLI Changes
+
+Our goal is to add opinionated commands that can setup a cluster of Kubernetes
+nodes as docker containers and bootstrap the bare minimum of services to
+successfully execute a `WorkflowRun` CRD tenantless. The new commands are
+opinionated in how they build, bootstrap and manage the workflow engine. We
+avoid the common pit-fall of over-engineering and reinventing the wheel by only
+providing configuration for the most necessary things to run Relay workflows.
+
+To handle the above, we will wrap and leverage
+[K3d](https://github.com/rancher/k3d) packages, which can bootstrap a working
+Kubernetes cluster inside docker containers.
+
+Clusters are singleton. There either is or isn't a relay development cluster on
+a machine. In V2, we will explore the ability to run more than 1 cluster by
+name, but this is notoriously complicated, so we are avoiding this complexity in
+V1.
+
+This works on MacOS and Linux as long as a running Docker service is present.
+
+#### Commands
+
+The following are commands to be added to the Relay CLI to support standalone
+clusters:
+
+`relay cluster start` - Start the cluster if it exists. Create and start the cluster
```

Comment:
How would you feel about moving this to something like `local-cluster`, `dev-cluster`, `dev cluster`, etc.? The problem I could see here is users downloading the CLI and worrying they need to run their own Kubernetes cluster to use Relay. @ahpook any opinions on this structure considering you wrote the initial CLI structure doc?